test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,460 @@
+# Owner(s): ["module: onnx"]
+
+"""Test consistency between torch.onnx exported operators and torch operators.
+
+Usage:
+
+    pytest test/onnx/test_op_consistancy.py
+
+    To run tests on a specific operator (e.g. torch.ceil):
+
+    pytest test/onnx/test_op_consistancy.py -k ceil
+
+Note:
+
+    When new ops are supported, please scroll down to modify the EXPECTED_SKIPS_OR_FAILS and
+    ALLOWLIST_OP lists. See "Modify this section"
+
+"""
+import contextlib
+import copy
+import dataclasses
+import io
+import itertools
+import unittest
+from typing import (
+    AbstractSet,
+    Callable,
+    Collection,
+    Iterable,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+import onnx
+
+import torch
+from torch.onnx import _constants, verification
+from torch.testing._internal import (
+    common_device_type,
+    common_methods_invocations,
+    common_utils,
+)
+from torch.testing._internal.opinfo import core as opinfo_core
+
+# The min onnx opset version to test for
+MIN_ONNX_OPSET_VERSION = 9
+# The max onnx opset version to test for
+MAX_ONNX_OPSET_VERSION = _constants.ONNX_MAX_OPSET
+
+TESTED_OPSETS = range(MIN_ONNX_OPSET_VERSION, MAX_ONNX_OPSET_VERSION + 1)
+ORT_PROVIDERS = ("CPUExecutionProvider",)
+
+SUPPORTED_DTYPES = (
+    # Boolean
+    torch.bool,
+    # Integers
+    torch.uint8,
+    torch.int8,
+    torch.int16,
+    torch.int32,
+    torch.int64,
+    # Floating types
+    torch.float16,
+    torch.float32,
+    torch.float64,
+    torch.bfloat16,
+    # QInt types
+    torch.qint8,
+    torch.quint8,
+    # Complex types
+    torch.complex32,
+    torch.complex64,
+    torch.complex128,
+)
+
+# Convenience tuples for creating dtype lists when skipping or xfailing tests
+
+BOOL_TYPES = (torch.bool,)
+
+INT_TYPES = (
+    torch.int8,
+    torch.int16,
+    torch.int32,
+    torch.int64,
+    torch.uint8,
+)
+
+QINT_TYPES = (
+    torch.qint8,
+    torch.quint8,
+)
+
+FLOAT_TYPES = (
+    torch.float16,
+    torch.float32,
+    torch.float64,
+    torch.bfloat16,
+)
+
+COMPLEX_TYPES = (
+    torch.complex32,
+    torch.complex64,
+    torch.complex128,
+)
+
+
+@dataclasses.dataclass
+class DecorateMeta:
+    """A dataclass for storing information about a test case to skip or xfail.
+
+    Adapted from functorch: functorch/test/common_utils.py
+    """
+
+    op_name: str
+    variant_name: str
+    decorator: Callable
+    device_type: Optional[str]
+    dtypes: Optional[Collection[torch.dtype]]
+    reason: str
+
+
+def xfail(
```

Comment:
`xfail`? Based on the comment below, shall we call it `should_fail`?

Reply:
This is based on other torch component's convention. Thoughts on following that convention?

Reply:
Sounds good.